tests/test_docker_images.py

```diff
@@ -0,0 +1,37 @@
+import docker
+from tests.helm_template_generator import render_chart
+import jmespath
+import pytest
+
+
+def list_docker_images():
+    extra_globals = {
+        "global.baseDomain": "foo.com",
+        "blackboxExporterEnabled": True,
+        "postgresqlEnabled": True,
+        "prometheusPostgresExporterEnabled": True,
+        "pspEnabled": True,
+        "veleroEnabled": True,
+    }
+
+    public_repo_docs = render_chart(values={"global": extra_globals})
+
+    # Listing docker images
+    search_string = "spec.template.spec.containers[*].image"
+    docker_images = []
+    for public_repo_doc in public_repo_docs:
+        docker_image_sublist = jmespath.search(search_string, public_repo_doc)
+
+        if docker_image_sublist is not None:
+            docker_images = docker_image_sublist + docker_images
+
+    return docker_images
+
+
+@pytest.mark.parametrize("docker_image", list_docker_images())
+def test_docker_image(docker_client, docker_image):
+    docker_image = docker_image.replace('"', "").strip()
+    try:
+        docker_client.images.get_registry_data(docker_image)
```

Comment:
Again replaced the `image->get` to `image->get_registry_data`. The `image->get` works for only local images. The  build was working fine for sometime when switched to `image->get` but started failing after may be because of caching on Circle CI side.

Reply:
Trying to test this, I see that it requires the docker daemon to be running. It would be great if we could find a way to skip these tests if docker daemon is not present.

I'm thinking something like:

```python
try:
    docker.from_env().ping()
    docker_daemon_present = True
except (docker.errors.DockerException):
    docker_daemon_present = False
```

and then on the test:

```
@pytest.mark.skipif(not docker_daemon_present, reason="Docker daemon not available")
```

Reply:
Done